Assets/MixedRealityToolkit.Services/CameraSystem/MixedRealityCameraSystem.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.CameraSystem
+{
+    /// <summary>
+    /// The Camera system controls the settings of the main camera.
+    /// </summary>
+    public class MixedRealityCameraSystem : BaseCoreSystem, IMixedRealityCameraSystem
+    {
+        private enum DisplayType
+        {
+            Opaque = 0,
+            Transparent
+        }
+
+        public MixedRealityCameraSystem(
+            IMixedRealityServiceRegistrar registrar,
+            BaseMixedRealityProfile profile = null) : base(registrar, profile)
+        {
+            cameraProfile = (MixedRealityCameraProfile)profile;
+        }
+
+        /// <summary>
+        /// Is the current camera displaying on an Opaque (AR) device or a VR / immersive device
+        /// </summary>
+        public bool IsOpaque
+        {
+            get
+            {
+                currentDisplayType = DisplayType.Opaque;
+#if UNITY_WSA
+                if (!UnityEngine.XR.WSA.HolographicSettings.IsDisplayOpaque)
+                {
+                    currentDisplayType = DisplayType.Transparent;
+                }
+#endif
+                return currentDisplayType == DisplayType.Opaque;
+            }
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; } = 0;
```

Comment:
What does SourceID mean here and why is it always zero

Reply:
It has to do with being a data provider. Core services are expected to be data providers by default, even if they're never expected to provide data. When they don't expect to provide data they set their SourceID to 0. Not sure what motivates this - @davidkline-ms may be able to weigh in.